src/plugins/intel_cpu/src/nodes/gather.cpp

```diff
@@ -179,6 +179,33 @@ void Gather::createPrimitive() {
     if (isInPlace()) {
         return;
     }
+
+    // short 1D vector fast executor (typical in shape infer subgraph)
+    execSpecialCase = nullptr;
+    if ((dataSrcRank == 1) && isDataShapeStat && isAxisInputConst && !constIndices.empty() && dataTypeSize == 4) {
```

Comment:
Why not to move this code on prepareParams stage to enable 1D impl even for dynamic data shape?

Reply:
sure, done.